lib/token.js

```diff
@@ -576,11 +575,22 @@ function prepareOauthParams(sdk, oauthOptions) {
 
 function getWithRedirect(sdk, oauthOptions, options) {
   oauthOptions = util.clone(oauthOptions) || {};
-  if (!oauthOptions.responseMode) {
-    oauthOptions.responseMode = 'fragment';
-  }
+
   return prepareOauthParams(sdk, oauthOptions)
     .then(function(oauthParams) {
+
+      // Dynamically set the responseMode unless the user has provided one
+      // Server-side flow requires query. Client-side apps usually prefer fragment.
+      if (!oauthOptions.responseMode) {
+        if (oauthParams.responseType.includes('code') && oauthParams.grantType !== 'authorization_code') {
+          // server-side flows using authorization_code
+          oauthParams.responseMode = 'query';
```

Comment:
nit: `query` is the default response mode for the `code` response type. According to the spec: `This use of this parameter is NOT RECOMMENDED with a value that specifies the same Response Mode as the default Response Mode for the Response Type used.`

Not sure that this can break something on BE side, just wanted to point out what spec says

Reply:
Just thinking out loud: according to OIDC spec `response_mode` is the optional parameter. Why sdk should make decision for the default value? Do we really need to set the value if it was not provided by the user?

Reply:
There is an existing default of `okta_message` that exists for the (also default) iframe flow. In this case, within `getWithRedirect` we must make a choice because `okta_message` will not be valid.

Reply:
One issue that I see with the code.
Let's say I want to use response type `code` and `pkce=true` flow. I don't set `responseMode` in config on purpose, because I know that default response mode for the flow I I want is `query`. However sdk sets `fragment` which is not oidc compliant behavior

Reply:
Our auth-js library currently only supports `fragment`. These are some good points, about OIDC defaults, I think we should take a closer look at this and possibly refactor.

Reply:
I've added a JIRA to track this work: https://oktainc.atlassian.net/browse/OKTA-243439

Reply:
Cool. Thank you! ;)